posts/2021-04-16-mp-context-propagration-21004.adoc

```diff
@@ -0,0 +1,353 @@
+
+To enable the MicroProfile Context Propagation 1.2 feature, add the following to your server configuration:
+[source, xml]
+----
+<featureManager>
+  <feature>mpContextPropagation-1.2</feature>
+  <!-- other features used by example code... -->
+  <feature>servlet-4.0</feature>
+  <feature>jdbc-4.2</feature>
+  <feature>jndi-1.0</feature>
+</featureManager>
+----
+
+Example usage within a Servlet:
+[source, java]
+----
+private ManagedExecutor executor;
+
+public void init(ServletConfig config) throws ServletException {
+    executor = ManagedExecutor.builder()
+                .propagated(ThreadContext.APPLICATION)
+                .cleared(ThreadContext.ALL_REMAINING)
+                .build();
+}
+
+public void destroy() {
+    executor.shutdownNow();
+}
+
+public void doGet(HttpServletRequest req, HttpServletResponse resp)
+    throws ServletException, IOException {
+    ...
+    executor.copy(unmanagedCompletionStage).thenAcceptAsync(value -> {
+        // requires java:comp namespace of the application,
+        DataSource ds = InitialContext.doLookup("java:comp/env/jdbc/ds");
+        ...
+    });
+}
+----
+
+For more information please see: 
+* link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.2/microprofile-context-propagation-spec-1.2.html[MicroProfile Context Propagation 1.2 specification]
+* link:https://download.eclipse.org/microprofile/microprofile-context-propagation-1.2/apidocs/[JavaDoc]
+
+[#ldapconnection]
+== LDAP connection supports Kerberos authentication
+
+LDAP bind operations are used to authenticate clients (and the users or applications behind them) to the directory server. This establishes an authorization identity that is used for subsequent operations that are processed on that connection, and specifies the LDAP protocol version that the client uses. Before this update, the `LdapRegistry` element supported binding either anonymously or by using simple authentication with a user (`bindDN`) and password (`bindPassword`). This update adds an option to bind to LDAP: GSSAPI/Kerberos. Kerberos is an authentication mechanism that allows a client to authenticate by a Key Distribution Center (KDC). In Open Liberty 21.0.0.4-beta,  you can use either a Kerberos credential cache (ccache) or a Kerberos keytab file.
```

Comment:
"In Open Liberty 21.0.0.4, you can use either a Kerberos credential cache (ccache) or a Kerberos keytab file."

aka, not 21.0.0.4-beta

Reply:
Done